modules/btcpayserver.nix

```diff
@@ -0,0 +1,225 @@
+    ];
+
+    systemd.services.nbxplorer = {
+      description = "Run nbxplorer";
+      path  = with pkgs; [
+        nix-bitcoin.nbxplorer
+        which
+        dotnet-sdk_3 ];
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "bitcoind.service" ];
+      after = [ "bitcoind.service" ];
+      preStart = ''
+        cp ${nbxplorer-configFile} ${cfg.nbxplorer.dataDir}/settings.config
+        chown -R '${cfg.nbxplorer.user}:${cfg.nbxplorer.group}' '${cfg.nbxplorer.dataDir}'
+        chmod 600 ${cfg.nbxplorer.dataDir}/settings.config
+        echo "btcrpcpassword=$(cat ${config.nix-bitcoin.secretsDir}/bitcoin-rpcpassword-btcpayserver)" >> '${cfg.nbxplorer.dataDir}/settings.config'
+      '';
+      postStart = ''
+        while [[ ! -e ${cfg.nbxplorer.dataDir}/Main/.cookie ]]; do
+          sleep 0.1
+        done
+        chmod 640 ${cfg.nbxplorer.dataDir}/Main/.cookie
+      '';
+      serviceConfig = nix-bitcoin-services.defaultHardening // {
+        ExecStart = "${cfg.nbxplorer.package}/bin/NBXplorer --conf=${cfg.nbxplorer.dataDir}/settings.config --datadir=${cfg.nbxplorer.dataDir}";
+        User = cfg.nbxplorer.user;
+        Restart = "on-failure";
+        RestartSec = "10s";
+        ReadWritePaths = cfg.nbxplorer.dataDir;
+        MemoryDenyWriteExecute = "false";
+      } // (if cfg.nbxplorer.enforceTor
+          then nix-bitcoin-services.allowTor
+          else nix-bitcoin-services.allowAnyIP
+        );
+    };
+
+    systemd.services.btcpayserver = {
+      description = "Run btcpayserver";
+      path  = with pkgs; [
+        nix-bitcoin.btcpayserver
+        dotnet-sdk_3 ]
+      ++ (optionals (cfg.btcpayserver.lightning-node == "lnd" ) [ openssl xxd ]);
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "nbxplorer.service" ]
+      ++ (optionals (cfg.btcpayserver.lightning-node == "clightning" ) [ "clightning.service" ])
+      ++ (optionals (cfg.btcpayserver.lightning-node == "lnd" ) [ "lnd.service" ]);
+      after = [ "nbxplorer.service" ]
+      ++ (optionals (cfg.btcpayserver.lightning-node == "clightning" ) [ "clightning.service" ])
+      ++ (optionals (cfg.btcpayserver.lightning-node == "lnd" ) [ "lnd.service" ]);
+      preStart = ''
+        cp ${btcpayserver-configFile} ${cfg.btcpayserver.dataDir}/settings.config
+        chown -R '${cfg.btcpayserver.user}:${cfg.btcpayserver.group}' '${cfg.btcpayserver.dataDir}'
+        chmod 600 ${cfg.btcpayserver.dataDir}/settings.config
+        ${optionalString (cfg.btcpayserver.lightning-node == "lnd") ''
+          certthumbprint=$(openssl x509 -noout -fingerprint -sha256 -in ${toString config.nix-bitcoin.secretsDir}/lnd-cert | sed -e 's/.*=//;s/://g')
+          echo -e "btclightning=type=lnd-rest;server=https://${toString cfg.lnd.listen}:${toString cfg.lnd.restPort}/;macaroonfilepath=${toString cfg.lnd.dataDir}/invoice+readonly.macaroon;certthumbprint=$certthumbprint" >> ${cfg.btcpayserver.dataDir}/settings.config
```

Comment:
I think you forgot dot? `btc.lightning`

Reply:
No, there is no dot
```
--btclightning              Easy configuration of lightning for the server administrator: Must be a UNIX socket of c-lightning (lightning-rpc) or URL to a charge server (default: empty)
```

Reply:
Ah, you're passing it on the command line, not in the config file.

Reply:
No, we're not. But I assumed it follows the same naming conventions. Apparently BTCPayServer is forgiving when it comes to this because the configuration option works.